### PR TITLE
Enable CodeQL Security Scan

### PR DESCRIPTION
**Pull request by VinodAnandan**
https://github.blog/2020-09-30-code-scanning-is-now-available/

https://github.com/quarkusio/quarkus/issues/11247

### CODE REVIEW

**Comment by sberyozkin** [comment]
@VinodAnandan  My concern is that we'll start getting a lot of false alarms.  Where are the results reported to ?

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        # By default, queries listed here will override any specified in a config file. 
+        # Prefix the list here with "+" to use these queries and those in the config file.
+        # queries: ./path/to/local/query, your-org/your-repo/queries@main
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+
+    - if: matrix.language == 'java' 
+      name: Build Java
+      run: mvn -B --settings .github/mvn-settings.xml clean install -DskipTests=true
```
Comment (gsmet) [inline .github/workflows/codeql-analysis.yml:57]: We also want `-DskipITs`

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+name: "CodeQL Security Scan"
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [master]
+  schedule:
+    - cron: '0 16 * * 4'
```
Comment (gsmet) [inline .github/workflows/codeql-analysis.yml:15]: Maybe let's do it on Sunday when the activity is low.

**Review by gsmet — CHANGES_REQUESTED** [review]
I have no idea what the reports look like but I think we can probably give it a try.

TBH, I was considering setting up SonarCloud for pull requests.

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (sberyozkin) [reply .github/workflows/codeql-analysis.yml:15]: And every 2nd Sunday for example ?

**Comment by VinodAnandan** [comment]
@sberyozkin @gsmet Thank you very much for reviewing my pull request. The CodeQL only focuses on the security issues unlike the sonarcloud and there are not that many findings. Most of the findings are related to the test code.

I believe the findings access will be restricted based on repo permission. You can access them by visiting this URL  " https://github.com/quarkusio/quarkus/security/code-scanning "  or navigate to the "Security" tab and then "Code scanning alerts"

I have modified to weekly (Sunday) scan and I think it should be minimum. I am happy to help to enable the sonarcloud scans for the pull requests.

Could you please review it again?

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Squashed the commits and merged, thanks!

I'll check the report tomorrow.

**Pull request merged**